src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/ReflectionEmitMemberAccessor.cs

```diff
@@ -38,7 +39,7 @@ internal sealed class ReflectionEmitMemberAccessor : MemberAccessor
 
             ILGenerator generator = dynamicMethod.GetILGenerator();
 
-            if (realMethod == null)
+            if (constructorInfo is null)
             {
```

Comment:
> { [](http://example.com/codeflow?start=11&length=2)

could you add assert here this is a value type? just to protect against future changes in this code.

Reply:
Done.